service/src/java/org/apache/hive/service/cli/OperationHandle.java

```diff
@@ -53,14 +53,19 @@ public void setHasResultSet(boolean hasResultSet) {
   }
 
   public boolean hasResultSet() {
-    return hasResultSet;
+    // null check to retain the old behavior
```

Comment:
No change needed in OperationHandle.java as we still return false if not set. Also, new method isHasResultSetIsSet is unused.